pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1543,15 +1543,36 @@ mod tests {
             std::fs::write(timeline_path.join(filename.file_name()), content).unwrap();
         }
 
+        let layer_file_1 = Layer::for_resident(
```

Comment:
I don't care about the repetition, can be polished in a future PR.

What do you mean by "metadata now passed twice"?

Reply:
I don't think fixing these tests *more* in this PR or preliminary PRs is important.

Reply:
by passed twice I mean: we first pass it here for `Layer::for_resident`, and then pass the metadata again to `schedule_layer_file_upload`. I'm not complaining about anything that is not introduced by this PR here.

Reply:
Ok, how do you see how this could be made better?

The reason why we need to remind about the metadata "now" so often is that I've aligned it to work with it on #5074 so the `for_resident` will also it instead of a `file_size` directly.

It is extra that `schedule_layer_file_upload` takes in a metadata instead of reading it from within the layer. I think I left it in place because other APIs were like this, let me confirm.

Reply:
sth like `for (layer_file_name, content) in layer_file_names[0..2].iter().zip(contents.iter()) { let layer_metadata = ...; let layer_file = ...; client.schedule_layer_file_upload(...); }`.

Reply:
Will soon push df328758f

Reply:
df328758f cleaned these up.